### ISSUE TITLE
Handling permissions errors on file I/O

### ISSUE THREAD

**Issue by khawkins** [shelljs/shelljs#64]
It would be nice if shelljs gracefully handled file I/O errors due to permissions issues, as opposed to the current full error dump you get now:

```
    var shelljs = require('./shelljs');
    shelljs.mkdir('-p', '/var/log/myNewDir');

    $ node shelljsConsumer.js
    shell.js: internal error
    Error: EACCES, permission denied '/var/log/myNewDir'
        at Object.fs.mkdirSync (fs.js:642:18)
        at mkdirSyncRecursive (shell.js:1649:8)
        at shell.js:541:7
        at Array.forEach (native)
        at Object._mkdir (shell.js:526:8)
        at Object.mkdir (shell.js:1491:23)
        at Object.<anonymous> (shelljsConsumer.js:2:10)
        at Module._compile (module.js:456:26)
        at Object.Module._extensions..js (module.js:474:10)
        at Module.load (module.js:356:32)
```

It seems like, for a lot of use cases (like this one), a more user-friendly error handling path could be initiated.

**Comment by akras14** [comment]
`mv` has a similar issue.

```
var shell = require("shelljs");
shell.mv("test", "/");

$: node test.js 
shell.js: internal error
Error: EACCES, permission denied 'test'
    at Object.fs.renameSync (fs.js:554:18)
    at /home/alexk/projects/support/uploader/node_modules/shelljs/src/mv.js:77:8
    at Array.forEach (native)
    at Object._mv (/home/alexk/projects/support/uploader/node_modules/shelljs/src/mv.js:53:11)
    at Object.mv (/home/alexk/projects/support/uploader/node_modules/shelljs/src/common.js:186:23)
    at Object.<anonymous> (/home/alexk/projects/support/uploader/test.js:3:7)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
```

In comparison `exec` seems to handle it gracefully, so I am switching to exec for my purposes.

```
var shell = require("shelljs");
shell.exec("mv test /");

$: node test.js 
mv: cannot move ‘test’ to ‘/test’: Permission denied
```

**Comment by nfischer** [comment]
#474 fixed this issue for `mkdir()`. Would anyone want to open a PR for `mv()`?

**Comment by nfischer** [comment]
This exception just means "there's a bug in shelljs, it didn't handle something it should have handled." So, this issue is about a very general problem.

I already fixed the mkdir issue in #474, and I filed an issue specific to mv in #872. I'm closing this in favor of the mv-specific issue.

---

Anyone stumbling across this thread: if you have a similar exception, please file a new github issue, as it will likely require a unique patch.
